gdlsc.go

```diff
@@ -1,104 +1,94 @@
 package main
 
 import (
-	"encoding/json"
 	"fmt"
-	"io/ioutil"
-	"log"
-	"net/http"
 	"os"
-	"os/exec"
+	"path"
+	"path/filepath"
 	"strings"
+
+	"github.com/ryanuber/go-license"
 )
 
-// URL represents path a github repos license
-var URL = "https://api.github.com/repos/%s/license?access_token=" + os.Getenv("ACCESS_TOKEN")
+const root = "vendor/"
 
-//types for message to be printed
-type message struct {
-	Dependency string
-	License    string
-	Link       string
-}
+var files []string
+var rootPath string
+var noLicense = map[string]bool{}
 
-// options for response from github
-type response struct {
-	Html_URL string
-	License  license
+// Response contains license and file info
+type Response struct {
+	License string
+	File    string
 }
 
-type license struct {
-	Name string
+func getLicense(input chan string, output chan Response) {
+	for file := range input {
+		l, err := license.NewFromFile(file)
+		splitFile := strings.Split(file, "/")
+		splitFile = splitFile[1 : len(splitFile)-1]
+		file := path.Join(splitFile...)
+		if err != nil {
+			output <- Response{File: file, License: err.Error()}
+		} else {
+			output <- Response{File: file, License: l.Type}
+		}
+	}
+}
+func checkLicense(path string) bool {
+	path = strings.ToUpper(path)
```

Comment:
Please use path.Base to avoid weird cases where there is an ancestor folder that contains LICENSE in the name. It's very unlikely but possible.